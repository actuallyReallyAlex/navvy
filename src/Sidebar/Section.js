import React from 'react'
import PropTypes from 'prop-types'
import { Box } from 'grommet'
import Header from './Header'
import Link from './Link'

const Section = ({ heading, links, selected, setSelected, size }) => {
  return (
    <Box margin={{ vertical: 'small' }} pad={{ vertical: 'small' }}>
      <Header heading={heading} size={size} />
      {links.map(({ collapsible, children, handler, icon, title }, i) => (
        <Link
          collapsible={collapsible}
          children={children}
          handler={handler}
          icon={icon}
          key={i}
          selected={selected}
          setSelected={setSelected}
          size={size}
          title={title}
        />
      ))}
    </Box>
  )
}

Section.propTypes = {
  heading: PropTypes.string.isRequired,
  links: PropTypes.array.isRequired,
  selected: PropTypes.string.isRequired,
  setSelected: PropTypes.func.isRequired,
  size: PropTypes.oneOf(['expanded', 'collapsed'])
}

export default Section
