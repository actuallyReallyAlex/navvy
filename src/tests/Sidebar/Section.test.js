import React from 'react'
import Section from '../../Sidebar/Section'
import { shallow } from 'enzyme'

const defaultProps = {
  heading: 'Heading',
  links: [
    { title: 'Form' },
    { title: 'Form Picker' },
    { title: 'Form Select' },
    { title: 'Form Masking' },
    { title: 'Form Wizard' }
  ],
  selected: 'Other',
  setSelected: jest.fn(),
  size: 'expanded'
}

describe('<Section />', () => {
  afterEach(() => {
    defaultProps.setSelected.mockClear()
  })

  test('Should render the <Section /> component.', () => {
    const component = shallow(<Section {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
