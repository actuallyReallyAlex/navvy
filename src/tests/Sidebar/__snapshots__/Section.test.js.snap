// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> Should render the <Section /> component. 1`] = `ShallowWrapper {}`;
